app/src/main/java/app/tivi/home/HomeActivity.kt

```diff
@@ -18,35 +18,30 @@ package app.tivi.home
 
 import android.content.Intent
 import android.os.Bundle
+import androidx.activity.viewModels
 import androidx.databinding.DataBindingUtil
 import androidx.lifecycle.observe
 import androidx.navigation.NavController
+import app.tivi.AppNavigator
 import app.tivi.R
-import app.tivi.TiviActivityMvRxView
-import app.tivi.account.AccountUiFragment
+import app.tivi.TiviActivity
 import app.tivi.databinding.ActivityHomeBinding
 import app.tivi.extensions.hideSoftInput
 import app.tivi.extensions.setupWithNavController
 import app.tivi.trakt.TraktConstants
-import com.airbnb.mvrx.viewModel
-import net.openid.appauth.AuthorizationException
-import net.openid.appauth.AuthorizationResponse
-import net.openid.appauth.AuthorizationService
+import dagger.hilt.android.AndroidEntryPoint
 import javax.inject.Inject
 
-class HomeActivity : TiviActivityMvRxView() {
-    private val authService by lazy(LazyThreadSafetyMode.NONE) {
-        AuthorizationService(this)
-    }
-
-    private val viewModel: HomeActivityViewModel by viewModel()
-
-    @Inject @JvmField var homeNavigationViewModelFactory: HomeActivityViewModel.Factory? = null
+@AndroidEntryPoint
+class HomeActivity : TiviActivity() {
+    private val viewModel: HomeActivityViewModel by viewModels()
 
     private lateinit var binding: ActivityHomeBinding
 
     private var currentNavController: NavController? = null
 
+    @Inject @JvmField var navigator: AppNavigator? = null
```

Comment:
You can avoid using the bang operator or null-checking for injected fields knowing you'll access them at the right time if you declare them with `lateinit`:
`@Inject @JvmField lateinit var navigator: AppNavigator`

Reply:
Yep, unfortunately there's an issue with `lateinit` with the Kotlin Compiler version used by Compose. This is the only way which currently works 🤦